### PR TITLE
📄 Create translation model and basic API

### LINKED ISSUE #331: Set up backend to fetch translations
We need an API to fetch our translations!  This will involve setting up a mongo data model (discussion point: what should this structure be?), and a simple GET request to fetch translations for a specified language (French, Chinese, or Spanish).  The format of our data needs to essentially look like a JSON of "messages".

We may also want to create an [admin endpoint](https://github.com/hack4impact-uiuc/ymca/blob/master/api/src/api/admin.js) to add translations for a specific language, for our testing purposes later.

Translated messages are just a dictionary of "ids" to "translation".  You can look [here](https://github.com/lingohub/example-resource-files/blob/master/JSON/example.fr.json) for an example of a French translation file.

### PR DESCRIPTION
**Pull request by Feeshay**
## Status:
:rocket: Ready

## Description
* Create translation model
* Add a get request to return translation model with specified language
* Add an admin post request to create a new translation object
* Add an admin put request to edit/create a new message translation
* Link endpoints to frontend api (hopefully it works lol)

Resolves #331

## TODO
* Enforce unique languages for post requests?

## Screenshots
Post:
image

Get:
image

Put:
image

### CODE REVIEW

File: client/src/utils/api.js [hunk 1]
```diff
@@ -174,3 +180,54 @@ export const deleteResource = (id: string): ApiResponse<null> => {
       },
     );
 };
+
+export const getTranslationByLanguage = (
+  language: string,
+): ApiResponse<Translation> => {
+  const requestExtension = `/api/translation/?language=${language}`;
```
Comment (jackieo5023) [inline client/src/utils/api.js:187]: I think we can remove the extra `/` here before the question mark just to keep with better HTTP standards - so it's `/api/translation?language=${language}` (this should still work, I tested on Postman)

File: api/src/api/admin.js [hunk 2]
```diff
@@ -181,4 +182,37 @@ router.delete(
...
+    updatedTranslation.messages.set(req.body.key, req.body.message);
+    await updatedTranslation.save();
+    res.json({
+      code: 200,
+      message: `Successfully added ${req.body.language} translation for ${req.body.key}`,
+      success: true,
+      result: updatedTranslation,
+    });
+  }),
+);
```
Comment (jackieo5023) [inline api/src/api/admin.js:216]: One random nit here - you can get rid of the repeated `req.body...` code by doing something like this at the beginning
```
const { language, key, message} = req.body;
```

**Review by jackieo5023 — APPROVED** [review]
awesome job!!!!

**Review by Lam7150 — APPROVED** [review]
swag

**Pull request merged**